src/ReplaceReferencesCapableTrait.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Dhii\Config;
+
+use InvalidArgumentException;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\ContainerInterface;
+use Dhii\Util\String\StringableInterface as Stringable;
+use Psr\Container\NotFoundExceptionInterface;
+
+/**
+ * Trait ReplaceReferencesCapableTrait.
+ *
+ * Adds ability to replace all tokens wrapped with some delimiters in a string with corresponding values retrieved from a container.
+ *
+ * @since [*next-version*]
+ */
+trait ReplaceReferencesCapableTrait
+{
+    /**
+     * Replaces all tokens wrapped with some delimiters in a string with corresponding values retrieved from a container.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable  $input          Input string to find and replace references
+     * @param ContainerInterface $container      Container to retrieve values for replace.
+     * @param string|Stringable  $startDelimiter Starting delimiter of token for replace
+     * @param mixed              $default        String to replace reference if key not found in container.
+     * @param string|Stringable  $endDelimiter   Ending delimiter of token for replace
+     *
+     * @throws ContainerExceptionInterface Error while retrieving the entry from the container.
+     *
+     * @return string The resulting string.
+     */
+    protected function _replaceReferences($input, ContainerInterface $container, $default = null, $startDelimiter = '${', $endDelimiter = '}')
+    {
+        $input        = $this->_normalizeString($input);
+        $defaultValue = $default === null ? '' : $this->_normalizeString($default);
+
+        $startDelimiter = preg_quote($this->_normalizeString($startDelimiter), '/');
```

Comment:
I think that having to type the same delimiter 4 times is a good sign that it should be centralized somewhere...